### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by HannaVasylieva**
https://HannaVasylieva.github.io/react_list-of-goods-js/

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -14,45 +16,118 @@ export const goodsFromServer = [
...
+const SORT_FEILD_LENGTH = 'length';
+
+function getPrepearedGoods(goods, toDo, direction) {
+  const prepearedGoods = [...goods];
+
+  if (!toDo && direction) {
+    return prepearedGoods.reverse();
+  }
+
+  if (toDo) {
```
Comment (Viktor-Kost) [inline src/App.jsx:29]: no needs to add this condition
```suggestion
```

File: src/App.jsx [hunk 2]
```diff
@@ -14,45 +16,118 @@ export const goodsFromServer = [
...
+        case SORT_FEILD_LENGTH:
+          if (good1.length !== good2.length) {
+            return (direction)
+              ? good2.length - good1.length
+              : good1.length - good2.length;
+          }
+
+          return (direction)
+            ? good2.localeCompare(good1)
+            : good1.localeCompare(good2);
```
Comment (Viktor-Kost) [inline src/App.jsx:41]:
```suggestion
  return good1.length - good2.length;
```

File: src/App.jsx [hunk 3]
```diff
@@ -14,45 +16,118 @@ export const goodsFromServer = [
...
+
+        case SORT_FEILD_ALPH:
+          return (direction)
+            ? good2.localeCompare(good1)
+            : good1.localeCompare(good2);
```
Comment (Viktor-Kost) [inline src/App.jsx:46]:
```suggestion
return good1.localeCompare(good2);
```

File: src/App.jsx [hunk 4]
```diff
@@ -14,45 +16,118 @@ export const goodsFromServer = [
...
+        </button>
+
+        {(sortFeild !== '' || sortReverse) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortFeild('');
+              setSortReverse(false);
+            }}
```
Comment (Viktor-Kost) [inline src/App.jsx:114]: move handler to an own function

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

File: src/App.jsx [hunk 5]
```diff
@@ -14,45 +16,107 @@ export const goodsFromServer = [
...
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-info',
```
Comment (etojeDenys) [inline src/App.jsx:61]:
```suggestion
            'button is-info',
```

**Review by etojeDenys — CHANGES_REQUESTED** [review]
change it everywhere

**Review by mykhalenych — APPROVED** [review]

**Pull request closed without merging**